tests/test_wsireader.py

```diff
@@ -1288,11 +1374,177 @@ def test_jp2_missing_cod(_sample_jp2):
         _ = wsi.info
 
 
+# @pytest.mark.skip(reason="working, skip to run other test")
+def test_read_bounds_location_in_requested_resolution(
+        _mini_wsi2_jp2,
+        _mini_wsi1_svs,
+        _mini_wsi1_jpg,
+        _mini_wsi1_msk):
+    """Actually a duel test for sync read and read at requested."""
+    # """Test synchronize read for VirtualReader"""
+    _mini_wsi2_jp2 = pathlib.Path(_mini_wsi2_jp2)
+    _mini_wsi1_svs = pathlib.Path(_mini_wsi1_svs)
+    _mini_wsi1_msk = pathlib.Path(_mini_wsi1_msk)
+    _mini_wsi1_jpg = pathlib.Path(_mini_wsi1_jpg)
+    # exit()
+    # root_dir = '/home/tialab-dang/workstation_storage_1/' \
+    #             'workspace/tiatoolbox/tests/local_samples/'
+    # _mini_wsi2_jp2 = pathlib.Path('%s/mini_wsi2.jp2'% root_dir)
+    # _mini_wsi1_svs = pathlib.Path('%s/CMU-mini.svs' % root_dir)
+    # _mini_wsi1_msk = pathlib.Path('%s/CMU-mask.png' % root_dir)
+    # _mini_wsi1_jpg = pathlib.Path('%s/CMU-mini.jpg' % root_dir)
+
+    def compare_reader(reader1, reader2, read_coord, read_cfg, check_content=True):
+        """Correlation test to compare output of 2 readers."""
+        requested_size = read_coord[2:] - read_coord[:2]
+        requested_size = requested_size[::-1]  # XY to YX
+        roi1 = reader1.read_bounds(
+                    read_coord,
+                    location_is_at_requested=True,
+                    pad_constant_values=255,
+                    **read_cfg,
+                )
+        roi2 = reader2.read_bounds(
+                    read_coord,
+                    location_is_at_requested=True,
+                    pad_constant_values=255,
+                    **read_cfg,
+                )
+        # ~ sane check code for local debug
+        import matplotlib.pyplot as plt
+        plt.subplot(1, 2, 1)
+        plt.imshow(roi1)
+        plt.subplot(1, 2, 2)
+        plt.imshow(roi2)
+        plt.savefig('dump.png')
+        plt.close()
+        # ~
+        assert roi1.shape[0] == requested_size[0], (
+                    read_cfg, requested_size, roi1.shape)
+        assert roi1.shape[1] == requested_size[1], (
+                    read_cfg, requested_size, roi1.shape)
+        assert roi1.shape[0] == roi2.shape[0], (read_cfg, roi1.shape, roi2.shape)
+        assert roi1.shape[1] == roi2.shape[1], (read_cfg, roi1.shape, roi2.shape)
+        if check_content:
+            cc = np.corrcoef(roi1[..., 0].flatten(), roi2[..., 0].flatten())
+            # this control the harshness of similarity test, how much should be?
+            assert np.min(cc) > 0.90, (cc, read_cfg, read_coord)
+
+    # * now check sync read by comparing the RoI with different base
+    # the output should be at same resolution even if source is of different base
+    msk = imread(_mini_wsi1_msk)
+    msk_reader = VirtualWSIReader(msk)
```

Comment:
we should always be using get_wsireader unless it is a tif.

Reply:
Similar to https://github.com/TIA-Lab/tiatoolbox/blob/ac77bdc8bbf73150178286e8e68af148d9372cc6/tests/test_wsireader.py#L1174. 
These tests are specifically for VRReader so I dont think using `get_wsireader` is informative when we read the test again in the future.